tmail-backend/mailbox/encrypted/api/src/main/scala/com/linagora/tmail/encrypted/event/EncryptedProjectionCleanupListener.java

```diff
@@ -0,0 +1,58 @@
+package com.linagora.tmail.encrypted.event;
+
+import javax.inject.Inject;
+
+import org.apache.james.events.Event;
+import org.apache.james.events.EventListener.ReactiveGroupEventListener;
+import org.apache.james.events.Group;
+import org.apache.james.mailbox.events.MailboxEvents.Expunged;
+import org.apache.james.mailbox.model.MessageMetaData;
+import org.reactivestreams.Publisher;
+
+import com.linagora.tmail.encrypted.EncryptedEmailContentStore;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+public class EncryptedProjectionCleanupListener implements ReactiveGroupEventListener {
+
+    public static class EncryptedProjectionCleanupGroup extends Group {
+    }
+
+    static final Group GROUP = new EncryptedProjectionCleanupGroup();
+
+    private final EncryptedEmailContentStore encryptedEmailContentStore;
+
+    @Inject
+    public EncryptedProjectionCleanupListener(EncryptedEmailContentStore encryptedEmailContentStore) {
+        this.encryptedEmailContentStore = encryptedEmailContentStore;
+    }
+
+    @Override
+    public Group getDefaultGroup() {
+        return GROUP;
+    }
+
+    @Override
+    public boolean isHandling(Event event) {
+        return event instanceof Expunged;
+    }
+
+    @Override
+    public Publisher<Void> reactiveEvent(Event event) {
+        if (event instanceof Expunged) {
+            Expunged expunged = (Expunged) event;
+
+            return handleMessageDeletion(expunged);
+        }
```

Comment:
We need to handle mailbox deletion too

Reply:
How can we get messageId list when mailbox deletion? 
CassandraMessageIdDAO?

Reply:
:+1:

Reply:
(We don't have much choice but this makes the listener specific to Cassandra. I would be happy that we abstract the use of the DAO behind an interface to be honnest...)

Reply:
> we abstract the use of the DAO behind an interface

I don't really understand what you mean. Can you explain in detail?